src/librustc_macros/src/hash_stable.rs

```diff
@@ -0,0 +1,87 @@
+use synstructure;
+use syn::{self, Meta, NestedMeta};
+use proc_macro2::{self, Ident, Span};
+
+struct Attributes {
+    ignore: bool,
+    project: Option<String>,
```

Comment:
If you made this an `Ident`, would everything still work out? Then diagnostics would directly link to the right place.